Automattic/wp-calypso
# remove /new flow for existing users. Simplify logic.

#### Changes proposed in this Pull Request

Now that the experiments are completed and we've decided to remove the /new flow for existing users pbxNRc-1jD-p2#comment-3163 we can redirect all (wpcom) users to the start flow.

Since we no longer rely on the `state` or on the abacus test assignment in order to determine the onboarding url, we can greatly simplify the code. 

We can not remove the entire `gutenboarding` project yet as it is still in use by anchor.fm paYKcK-1rX-p2. 

#### Implementation notes

I chose to move get-onboarding-url to signup/config since it no longer is a selector on `state`. I also removed `useOnboardingUrl` and the provider as we can now consolidate on one simple function. I finished the tidy up by inlining getOnboardingUrl and removing gutenboarding_url from config.

I see that gutenboarding_url is still made available in the config of the inline-help and happychat apps. I'm not familiar with these apps and can't work out if/how it's being used so chose to leave it in for now just to be safe and limit the amount of testing we need to do.

#### Testing instructions
There are a number of ways you can see the "Add new site" link.
The simplest is to open the "switch sites" menu and then click "add new site" at the bottom of the screen
If you search code for `getOnboardingUrl` you'll see that there are 7 other places where the onboarding url is used including help text and scenarios where the user has only one site or attempts to close their account.

Navigating to the `/new` endpoint should still work as before.
